pkg/cache/cache.go

```diff
@@ -33,25 +33,49 @@ import (
 	"k8s.io/klog/v2"
 )
 
-type CacheResponse struct {
+type Entry struct {
 	RoleARN         string
 	Audience        string
 	UseRegionalSTS  bool
 	TokenExpiration int64
 }
 
+type Request struct {
+	Name                string
+	Namespace           string
+	RequestNotification bool
+}
+
+func (r Request) CacheKey() string {
+	return r.Namespace + "/" + r.Name
+}
+func (r Request) WithNotification() Request {
+	r.RequestNotification = true
+	return r
+}
+
+type Response struct {
+	RoleARN         string
+	Audience        string
+	UseRegionalSTS  bool
+	TokenExpiration int64
```

Comment:
I agree, it would simplify the code here. However, it also exposes `Entry`, which as of now, is a cache internal datastructure ... however, it contains only fields `Response` is also having ... I'm really undecided.

What I definitely see is, that we don't need both,  `FoundInCMCache` and `FoundInSACache`. At the beginning when I started contributing to the project, I thought we would, but now I know better. I changed that into a single field.

Reply:
Maybe this is an option?
```
type Response struct {
	Entry
	Notifier        <-chan struct{}
}
```
This still exposes `Entry` to the outside world but only because it is exported for usage in `pkg/cache/debug`. Could be refactored one day? 
Because `Entry` is not a pointer in this setup, we can not do the proposed `if response.entry == nil`, however, I think returning an additional boolean indicating if found or not in `cache.Get(...)` is anyway more Go-like. We could implemented it like that. WDYT?

Reply:
It seems good enough as-is - probably not worth embedding the struct IMO. We can always DRY things up later.

Glad we were able to consolidate those flags :)